Source/resources/mongoDB/internal/MongoDBResource.ts

```diff
@@ -29,17 +32,24 @@ export class MongoDBResource extends IMongoDBResource {
         executionContext: ExecutionContext,
         logger: Logger) {
         super(ResourceName.from('MongoDB'), tenant, client, executionContext, logger);
-        this._method = client.getMongoDb;
-    }
-
-    /** @inheritdoc */
-    getConnectionString(cancellation?: Cancellation): Promise<MongoDBConnectionString> {
-        return this.get(this._method, response => MongoDBConnectionString.from(response.getConnectionstring()), cancellation);
+        this._method = client.getMongoDB;
     }
 
     /** @inheritdoc */
-    protected getResultFromResponse(response: GetMongoDbResponse): MongoDBConnectionString {
-        return MongoDBConnectionString.from(response.getConnectionstring());
+    async getDatabase(databaseSettingsCallback?: DatabaseSettingsCallback, cancellation?: Cancellation): Promise<Db> {
+        if (this._openClients.has(this.tenant.toString())) {
+            return this._openClients.get(this.tenant.toString())![1];
+        }
```

Comment:
Maybe we should just keep the MongoClient in the map - and create the DB every time this method is called? Right now the database settings callback is ignored after the first time. So if you provide different options - it won't work.

Reply:
That is true! Let's do that